builds/fabric8-che/plugins/keycloak-plugin-shared/src/main/java/com/redhat/che/keycloak/shared/KeycloakSettings.java

```diff
@@ -0,0 +1,48 @@
+package com.redhat.che.keycloak.shared;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Map;
+
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class KeycloakSettings {
+    
+    private static final Logger LOG = Logger.getLogger(KeycloakSettings.class.getName());
+
+    private static Map<String, String> settings = null;
+
+    public static Map<String, String> get() {
+        return settings;
+    }
+
+    public static void set(Map<String, String> theSettings) {
+        settings = theSettings;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static void pullFromApiEndpointIfNecessary(String apiEndpoint) {
+        if (settings == null) {
+            URL url;
+            HttpURLConnection conn;
+            try {
+                url = new URL(KeycloakConstants.getEndpoint(apiEndpoint));
+                LOG.info("Pulling Keycloak settings from URL :" + url);
```

Comment:
IMHO I think my point is only that this info is useless (only useful when you want to debug it). Not if the traces is printed a lot of time or not.